src/core/symbology-ng/qgsstylev2.cpp

```diff
@@ -947,6 +961,24 @@ int QgsStyleV2::getId( const QString& table, const QString& name )
   return id;
 }
 
+QString QgsStyleV2::getName( const QString& table, int id ) const
+{
+  char *query = sqlite3_mprintf( "SELECT name FROM %q WHERE id='%q'", table.toUtf8().constData(), QString::number( id ).toUtf8().constData() );
+
+  sqlite3_stmt *ppStmt;
+  int nErr = sqlite3_prepare_v2( mCurrentDB, query, -1, &ppStmt, nullptr );
+
+  QString name = "";
```

Comment:
Don't do this either ;) it means the literal has to be parsed into the string. Just using the default constructor has the same effect but is more efficient.